substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageBFDNameProvider.java

```diff
@@ -510,6 +521,16 @@ private void mangleType(ResolvedJavaType type) {
             }
         }
 
+        private void mangleType(UnresolvedJavaType type) {
+            if (type.isArray()) {
```

Comment:
We don't have a class loader for this unresolved type so I am not sure that we can guarantee that its name will not clash with a name for some valid resolved type. It might be safer to put all unresolved types into their own namespace to ensure this does not happen. That would also involve modifying the debug type info generation code to adopt the same namespace convention.

I'll leave that change for a follow-up PR once we can identify what other changes are required to ensure that debug info generator does not break if an unresolved type pops up as an element of the declaration closure for the runtime type set.